src/apps/ENGINE/s_functab.h

```diff
@@ -15,79 +16,84 @@
 // when segment_id is INTERNAL_SEGMENT_ID, this function is internal and offset
 // means internal function number
 
-struct LVARINFO
+struct LocalVarInfo
 {
+    LocalVarInfo() = default;
+
+    bool IsArray()
+    {
+        return elements > 1;
+    }
+
+    std::string name;
+    size_t hash; // for faster comparison
```

Comment:
can you explain how it is used? I cannot find it's by inspecting the commit...

Reply:
It is used in FuncTable::FindVar, [this](https://github.com/storm-devs/storm-engine/blob/2bea837a898bc16d6209318065fd48a2cdabc1bc/src/apps/ENGINE/s_functab.cpp#L178) line. The same thing was done in the old code.

Reply:
So it is hashed twice, right? I recall now that this was another point against stock unordered_map
Though it is still fine for now
Everything else is good! I think this can be merged after rebase

Reply:
Variable names in FuncTable are not actually hashed twice, because they are not stored in an unordered_map. They are stored in a vector (local_vars) which is stored in FuncInfo which is stored in an unordered_map. In order to find a functions' argument you have to look that function up first, using unordered_map, and then find the argument, using this argument's hash to speed up comparison. The same thing was done in the old code, the difference is that it used a custom hash table.

Reply:
Ah, that makes sense
I believe I tried to invent a container for string interning. I see now this is unchanged here